examples/rust/Earthfile

```diff
@@ -0,0 +1,13 @@
+FROM rust:1.24.0
+
+WORKDIR /usr/src/app
+
+build:
+    COPY . .
+    RUN cargo install
```

Comment:
Can the downloading of dependencies be moved to a separate target? (eg `deps`)

It helps show typical usage of Earthly with a real app, where the Earthly doesn't need to download all dependencies on each code change.